### ISSUE TITLE
Low accuracy for concrete_sctength dataset

### ISSUE THREAD

**Issue by StpMax** [mindsdb/mindsdb_native#298]
Last staging of all, no gpu

I get accuracy = 1% for column `concrete_strength` in `concrete_strength` dataset:
```
from mindsdb_native import FileDS, Predictor, F
ds = FileDS('benchmarks/datasets/concrete_strength/data.csv')
p = Predictor(name='cs')
p.learn(from_data=ds, to_predict=['concrete_strength'])
F.get_model_data('cs')['accuracy']
```

**Comment by paxcema** [comment]
Predictions look okay, so I guess the issue here is that the logic behind the reported 1% accuracy is faulty.

**Comment by maximlopin** [comment]
> Predictions look okay, so I guess the issue here is that the logic behind the reported 1% accuracy is faulty.

`ProbabilisticValidator` returns 0.01 accuracy, i'm trying to figure out why

**Comment by paxcema** [comment]
I ran some tests and it's the `AccStats.get_accuracy_stats()` method that is faulty. As it stands, I'd probably opt for reporting the accuracy directly from [`normal_accuracy`](https://github.com/mindsdb/mindsdb_native/blob/aa9346a340f4c1bc7641b0919f679f0e2c4278ef/mindsdb_native/libs/phases/model_analyzer/model_analyzer.py#L41).

**Comment by maximlopin** [comment]
With this (https://github.com/mindsdb/mindsdb_native/pull/321) fix the accuracy is ~0.3 but still seems wrong.

For some reason `confidence_range[0]` is sometimes greater than `confidence_range[1]`

**Comment by George3d6** [comment]
See https://github.com/mindsdb/mindsdb_native/issues/339 as a followup.
